### PR TITLE
Fix flaky unit tests

### PR DESCRIPTION
**Pull request by hoxbro**
They are flaky and currently impossible to recreate locally. 

Also increased number of reruns for UI test

This command `pytest panel/tests/ -k 'not vtk' -n 4 --dist loadgroup --randomly-seed=3306344094` seems to trigger the random failure somewhat regularly on main.

### CODE REVIEW

File: panel/tests/theme/test_base.py [hunk 1]
```diff
@@ -1,12 +1,16 @@
 import pathlib
 
+import pytest
+
```
Comment (hoxbro) [inline panel/tests/theme/test_base.py:4]:
```suggestion
```

File: panel/tests/theme/test_base.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 from bokeh.models import ImportedStyleSheet
 
 from panel.io.resources import CDN_DIST
 from panel.theme.base import BOKEH_DARK, Design, Inherit
 from panel.viewable import Viewable
 from panel.widgets import FloatSlider, IntSlider, TextInput
 
+pytestmark = pytest.mark.flaky(reruns=5, reruns_delay=10)
+
```
Comment (hoxbro) [inline panel/tests/theme/test_base.py:13]:
```suggestion
```

File: pixi.toml [hunk 3]
```diff
@@ -142,7 +142,7 @@ packaging = "*"
 _install-ui = 'playwright install chromium'
 
 [feature.test-ui.tasks.test-ui]
-cmd = 'pytest panel/tests/ui --ui --browser chromium -n logical --dist loadgroup --reruns 3 --reruns-delay 10'
+cmd = 'pytest panel/tests/ui --ui --browser chromium -n logical --dist loadgroup --reruns 5 --reruns-delay 20'
```
Comment (hoxbro) [inline pixi.toml:145]:
```suggestion
cmd = 'pytest panel/tests/ui --ui --browser chromium -n logical --dist loadgroup --reruns 3 --reruns-delay 10'
```

**Comment by hoxbro** [comment]
The last changes I have added makes this random run pass:  `pytest panel/tests/ -k 'not vtk' --randomly-seed=3186128261`

File: panel/tests/conftest.py [hunk 4]
```diff
@@ -42,6 +42,12 @@
 JUPYTER_TIMEOUT = 15 # s
 JUPYTER_PROCESS = None
 
+if os.name != 'nt':
+    import resource
+
+    soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+    resource.setrlimit(resource.RLIMIT_NOFILE, (hard, hard))
```
Comment (hoxbro) [inline panel/tests/conftest.py:49]: I think this is the culprit of many flaky failures in the CI.

File: panel/tests/util.py [hunk 5]
```diff
@@ -295,8 +301,9 @@ def serve_component(page, app, suffix='', wait=True, **kwargs):
 
 def serve_and_request(app, suffix="", n=1, port=None, **kwargs):
     port = serve_and_wait(app, port=port, **kwargs)
-    reqs = [requests.get(f"http://localhost:{port}{suffix}") for i in range(n)]
-    return reqs[0] if len(reqs) == 1 else reqs
+    reqs = [r for _ in range(n) if (r := requests.get(f"http://localhost:{port}{suffix}")).ok]
+    assert len(reqs) == n, "Not all requests were successful"
+    return reqs[0] if n == 1 else reqs
```
Comment (hoxbro) [inline panel/tests/util.py:306]: I just want to be more verbose here.

File: panel/tests/ui/template/test_editabletemplate.py [hunk 6]
```diff
@@ -176,7 +176,7 @@ def test_editable_template_drag_item(page):
 
     md2_handle.drag_to(md2_handle, target_position={'x': 0, 'y': -50}, force=True)
 
-    wait_until(lambda: list(tmpl.layout) == [id(md2), id(md1)], page)
+    wait_until((lambda: list(tmpl.layout), lambda: [id(md2), id(md1)]), page)
```
Comment (hoxbro) [inline panel/tests/ui/template/test_editabletemplate.py:179]: I'm testing this out as I want more information than currently given with `wait_until`

Comment (philippjfr) [reply panel/tests/ui/template/test_editabletemplate.py:179]: Yes, had a go at trying to get pytest to understand `wait_until` recently, without luck.

Comment (hoxbro) [reply panel/tests/ui/template/test_editabletemplate.py:179]: Also looked into it found this http://pybites.blogspot.com/2011/07/behind-scenes-of-pytests-new-assertion.html

File: .github/workflows/test.yaml [hunk 7]
```diff
@@ -145,6 +145,7 @@ jobs:
           opengl: true
       - name: Test unit
         run: |
+          ulimit -n || true
```
Comment (hoxbro) [inline .github/workflows/test.yaml:148]:
```suggestion
```

**Comment by hoxbro** [comment]
This combination fails around 1/10  of the time:
`pytest 'panel/tests/test_server.py::test_server_thread_pool_busy' 'panel/tests/test_server.py::test_server_session_info' -p 'no:randomly'`

**Comment by hoxbro** [comment]
Also, seeing this flaky failure locally

image

**Comment by hoxbro** [comment]
The generator tests in `test_param.py` also seem flaky.

**Comment by hoxbro** [comment]
I also sometimes have the following flaky test, when running with xdist
image

File: panel/tests/theme/test_base.py [hunk 8]
```diff
@@ -185,7 +200,11 @@ def test_design_apply_with_dark_theme(document, comm):
     model = widget.get_root(document, comm=comm)
     model.document = document
 
-    DesignTest(theme='dark').apply(widget, model)
+    dt = DesignTest(theme='dark').apply(widget, model)
```
Comment (hoxbro) [inline panel/tests/theme/test_base.py:203]:
```suggestion
    dt = DesignTest(theme='dark')
```

File: panel/tests/conftest.py [hunk 9]
```diff
@@ -42,11 +42,41 @@
...
+
+
+for e in os.environ:
+    if e.startswith(('BOKEH_', "PANEL_")) and e not in ("PANEL_LOG_LEVEL"):
```
Comment (hoxbro) [inline panel/tests/conftest.py:53]:
```suggestion
    if e.startswith(('BOKEH_', "PANEL_")) and e not in ("PANEL_LOG_LEVEL", ):
```

File: panel/reactive.py [hunk 10]
```diff
@@ -317,7 +317,8 @@ def _scheduled_update_model(
         try:
             self._update_model(events, msg, root, model, doc, comm)
         finally:
-            del self._in_process__events[doc]
+            if doc in self._in_process__events:
+                del self._in_process__events[doc]
```
Comment (hoxbro) [inline panel/reactive.py:321]:
```suggestion
            self._in_process__events.pop(doc, None)
```

This should be more threadsafe

**Pull request merged**